bot/exts/utilities/epoch.py

```diff
@@ -0,0 +1,123 @@
+from typing import Union
+
+import arrow
+import dateutil
+import discord
+from dateutil import parser
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.exts.core.extensions import invoke_help_command
+
+# https://discord.com/developers/docs/reference#message-formatting-timestamp-styles
+TIMESTAMP_FORMATS = [
+    "h:mm A",
+    "h:mm:ss A",
+    "MM/DD/YYYY",
+    "MMMM D, YYYY",
+    "MMMM D, YYYY h:mm A",
+    "dddd, MMMM D, YYYY h:mm A "
+]
+STYLES = {"Epoch": "",
+          "Short Time": "t",
+          "Long Time": "T",
+          "Short Date": "d",
+          "Long Date": "D",
+          "Short Date/Time": "f",
+          "Long Date/Time": "F",
+          "Relative Time": "R"
+          }
```

Comment:
This is pretty order dependent. What would you say about having `STYLES` hold a tuple containing the format and the identifier?

Reply:
I'm not sure what you mean here. What would that look like?

Reply:
A list of tuples would look like this:
```python
STYLES = [
	("Epoch", ""),
	("Short Time", "t"),
	("Long Time", "T"),
	...
]
```

Reply:
> A list of tuples would look like this:
> 
> ```python
> STYLES = [
> 	("Epoch", ""),
> 	("Short Time", "t"),
> 	("Long Time", "T"),
> 	...
> ]
> ```
I don't see the benefit over a dictionary. That should retain order as well, no?

Reply:
If my memory serves the correct information, dictionaries retain insertion order. So what you have now *should* be fine.

Reply:
No no, I was thinking about a dictionary of tuples, something like
```py
STYLES = {
    "Short time": ("t", "h:mm A"),
    "Long time": ("T", "h:mm:ss A")
    ...
}```
To remove `TIMESTAMP_FORMAT`

Reply:
Ah, yeah. That makes more sense. I agree with this suggestion.